scenariorun/src/main/kotlin/com/cosmotech/scenariorun/ContainerFactory.kt

```diff
@@ -1624,25 +1624,48 @@ internal fun getCommonEnvVars(
   return (minimalEnvVars + commonEnvVars).toMutableMap()
 }
 
+private fun getNewDependenciesIfExist(
+    mergedContainerNames: MutableMap<String, String>,
+    container: ScenarioRunContainer
+): ScenarioRunContainer {
+  var tmp: ScenarioRunContainer = container
+  var dependencies: MutableList<String> = mutableListOf()
```

Comment:
Can be rewrite as `val dependencies = mutableListOf<String>()`
 Plus, what append when you have same value several times? 
 How Argo deals with it? Could it be interesting to use `mutableSetOf<String>()` instead ?

Reply:
dependencies as list is a legacy code, ref ScenarioRunContainer.kt
In Argo's point of view, it expects a list(not a set) so it doesn't care of duplicated name i suppose